Source/Core/Core/IOS/Network/KD/Mail/WC24Send.h

```diff
@@ -31,11 +33,37 @@ class WC24SendList final
   void WriteSendList() const;
 
   std::string_view GetMailFlag() const;
+  u32 GetNumberOfMail() const;
+  std::vector<u32> GetMailToSend() const;
+  u32 GetEntryId(u32 entry_index) const;
+  u32 GetMailSize(u32 index) const;
+  ErrorCode DeleteMessage(u32 index);
+  std::string GetMailPath(u32 index) const;
+  u32 GetNextEntryId() const;
+  u32 GetNextEntryIndex() const;
+
+  ErrorCode AddRegistrationMessages(const WC24FriendList& friend_list, u64 sender);
 
 private:
   static constexpr u32 MAX_ENTRIES = 127;
   static constexpr u32 SEND_LIST_SIZE = 16384;
 
+  static constexpr char MAIL_REGISTRATION_STRING[] =
+      "MAIL FROM: {:16d}@wii.com\r\nRCPT TO: {:16d}wii.com\r\nDATA\r\nDate: Fri, 19 Nov 2021 "
+      "00:00:00 GMT\r\nFrom: "
+      "{:16d}wii.com\r\nTo: {:16d}wii.com\r\nMessage-Id: "
+      "lol\r\nSubject: WC24 Cmd Message\r\nX-Wii-AppId: 0-00000001-0001\r\nX-Wii-Cmd: "
+      "80010001\r\nMIME-Version: 1.0\r\nContent-Type: multipart/mixed;\r\n "
+      "boundary=\"Boundary-NWC24-041B6CE500012\"\r\n--Boundary-NWC24-041B6CE500012\r\nContent-Type:"
+      " text/plain; charset=us-ascii\r\nContent-Transfer-Encoding: 7bit\r\nWC24 Cmd "
+      "Message\r\n--Boundary-NWC24-041B6CE500012\r\nContent-Type: application/octet-stream;\r\n "
+      "name=a0000018.dat\r\nContent-Transfer-Encoding: base64\r\nContent-Disposition: "
+      "attachment;\r\n "
+      "filename=a0000018.dat\r\n\r\n "
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
+      "\r\n\r\n--Boundary-NWC24-041B6CE500012--";
```

Comment:
Please format this better, just make a manual linebreak on each `\r\n`. Also does it make sense to hardcode the Date here?

Reply:
Isn't this basically just 128 bytes of zero? Seems like a weird thing to send as payload.

I'm also curious if this is actually following the multipart specification and takes any boundary string, or if anything _actually_ validates the format.

Reply:
> Also does it make sense to hardcode the Date here?

It doesn't make much sense, I should probably change it; however it doesn't affect where the message would be placed in the message board (This won't appear on the message board anyways as it is a registration command message). 

> Isn't this basically just 128 bytes of zero?

Yes it is. Internally this is some type of user data which is wrote to the friend's entry in the `nwc24fl.bin`, however I have never seen this be anything other than 0.